### DISCUSSION TITLE
Which Altitude Value is Stored in the HERO11 Metadata?

### DISCUSSION THREAD

**Discussion by AlexLandherr** [gopro/labs#590] [General]
I'm curious as to which altitude value is used for the metadata stored by my HERO11 Black?

I've taken my glossary of terms from here:
[Elevation for beginners](http://eos-gnss.com/knowledge-base/articles/elevation-for-beginners)

Elevation types:
1. Ellipsoidal Height
2. Geoid Height
3. Orthometric Height

I am asking since I am using the [Telemetry Overlay](https://goprotelemetryextractor.com/telemetry-overlay-gps-video-sensors) tool for overlaying the recorded GPS data.

**Comment by dnewman-gpsw** [comment]
GPS9 metadata uses altitude above mean sea level.

**Reply by AlexLandherr** [reply]
So it would be this (from above link):
```
However, it’s important to note that the global MSL on your GPS receiver is generally based on a coarse 10-minute by 10-minute grid. This can make the global MSL elevations output by GPS receivers off by several meters.
```

**Reply by dnewman-gpsw** [reply]
Sure.

**Reply by AlexLandherr** [reply]
And it would as of writing this be impossible to get out the height above the WGS84 ellipsoid (which I assume is the reference ellipsoid used here) without completely changing the metadata format?

**Reply by dnewman-gpsw** [reply]
That it was WGS84 years ago, then there was request for MSL, being closer to real user experience. It is not user controllable. The metadata format is not affected as it supports both types.

**Reply by dnewman-gpsw** [reply]
Is this request for a future Labs control?  Can you explain why you prefer WGS84 data?

**Reply by AlexLandherr** [reply]
Thanks! Two more questions since I'm a sucker for technical details:

When you say "WGS84" do you mean GoPro cameras used the altitude relative to the ellipsoid used by the World Geodetic System in the past (of the camera firmware)?

Is the "MSL" value now used by GPS-capable GoPro's based on the [Earth Gravitational Model](https://en.wikipedia.org/wiki/Earth_Gravitational_Model?wprov=sfti1) used today by the World Geodetic System as its geoid?

**Reply by AlexLandherr** [reply]
> Is this request for a future Labs control?  Can you explain why you prefer WGS84 data?  

This is not necessarily a request for future labs control (but I would favor it if deemed worthwhile) of the option to use height above the WGS84 ellipsoid.

The reason I prefer WGS84 data is I like to have my geographical information's reference system known to myself so that any future calculations based on data from my GoPro footage can be done with more certainty.

**Reply by dnewman-gpsw** [reply]
This data all comes from u-blox, the GPS part vendor.  This particular implementation details is not known to me.  So the precision of MSL model is likely internal info to u-blox.  For implementation we got to choose between WGS84 and MSL.

**Reply by AlexLandherr** [reply]
Ok.

Would you and the other developers be opposed to a Labs feature that would let people choose (not via menu but via QR code) which altitude value is recorded?

Might be useful for pilots etc. to choose ellipsoidal height when MSL doesn't "cut it".

I understand if it's not worthwhile implementing it.

**Reply by dnewman-gpsw** [reply]
I've worked it out, I think, so it should be straight forward.

**Reply by AlexLandherr** [reply]
> I've worked it out, I think, so it should be straight forward.  

How to get ellipsoidal height recorded as an option to MSL? If so I'd gladly champion it.

**Reply by dnewman-gpsw** [reply]
Yes.  Command will be oMWG84=1

**Reply by AlexLandherr** [reply]
Could it be made permanent with an exclamation mark "!" instead of an "o"?

**Reply by dnewman-gpsw** [reply]
That's the way it all works.  I just share the non-permanent commands to prevent less Labs familiar users making permanent errors.

**Reply by AlexLandherr** [reply]
> That's the way it all works.  I just share the non-permanent commands to prevent less Labs familiar users making permanent errors.

Ok, that's understandable. Any approximate timeline when this would be enabled/available for HERO11 Black etc.?

**Reply by dnewman-gpsw** [reply]
It is out yesterday.  The command !MWG84=1 will permanently switch the altitude to the WGS84 ellipsoid.

**Reply by AlexLandherr** [reply]
Thanks so much!

I created this all in one link for the new command plus GPS Sync & 24Hz:
[GPS Sync & 24Hz & WGS84 Ellipsoidal Height](https://gopro.github.io/labs/control/set/?cmd=!MSYNC=1!M24HZ=1!MWG84=1&title=Permanent%20GPS%20Sync%20and%2024Hz%20and%20WGS84%20Ellipsoidal%20Height)

The new command shows as "WG84" on boot.

**Reply by dnewman-gpsw** [reply]
All active extension should be list on boot, as I reminder it is there.  All good.
